src/Collection.php

```diff
@@ -812,6 +893,24 @@ public function listIndexes(array $options = [])
         return $operation->execute(select_server($this->manager, $options));
     }
 
+    /**
+     * Returns information for all Atlas Search indexes for the collection.
+     * Only available when used against a 7.0+ Atlas cluster.
+     *
+     * @param array{name?: string} $options Command options
+     * @return Countable&Iterator<array{id: string, name: string, status: string, queryable: bool, latestDefinition: array}>
```

Comment:
The return type depends on the `typeMap` option. I have to relax the return type.